src/Compilers/Core/Portable/Serialization/ObjectWriter.cs

```diff
@@ -560,43 +532,7 @@ private void WriteArray(Array array)
             }
             else
             {
-                // emit header up front
-                this.WriteKnownType(elementType);
-
-                // recursive: write elements now
-                var oldDepth = _recursionDepth;
-                _recursionDepth++;
-
-                if (_recursionDepth % MaxRecursionDepth == 0)
-                {
-                    _cancellationToken.ThrowIfCancellationRequested();
-
-                    // If we're recursing too deep, move the work to another thread to do so we
-                    // don't blow the stack.
-                    var task = SerializationThreadPool.RunOnBackgroundThreadAsync(
-                        a =>
-                        {
-                            WriteArrayValues((Array)a!);
-                            return null;
-                        },
-                        array);
-
-                    // We must not proceed until the additional task completes. After returning from a write, the underlying
-                    // stream providing access to raw memory will be closed; if this occurs before the separate thread
-                    // completes its write then an access violation can occur attempting to write to unmapped memory.
-                    //
-                    // CANCELLATION: If cancellation is required, DO NOT attempt to cancel the operation by cancelling this
-                    // wait. Cancellation must only be implemented by modifying 'task' to cancel itself in a timely manner
-                    // so the wait can complete.
-                    task.GetAwaiter().GetResult();
```

Comment:
who remembers this awful code, to prevent stack overflowing in tree writing.  I do :)  Die code die.